src/java.base/share/classes/java/lang/constant/ClassDesc.java

```diff
@@ -158,19 +158,10 @@ static ClassDesc of(String packageName, String className) {
      * @see ClassDesc#ofInternalName(String)
      */
     static ClassDesc ofDescriptor(String descriptor) {
-        requireNonNull(descriptor);
-        if (descriptor.isEmpty()) {
-            throw new IllegalArgumentException(
-                    "not a valid reference type descriptor: " + descriptor);
-        }
-        int depth = ConstantUtils.arrayDepth(descriptor);
-        if (depth > ConstantUtils.MAX_ARRAY_TYPE_DESC_DIMENSIONS) {
-            throw new IllegalArgumentException(
-                    "Cannot create an array type descriptor with more than " +
-                    ConstantUtils.MAX_ARRAY_TYPE_DESC_DIMENSIONS + " dimensions");
-        }
+        // implicit null-check
         return (descriptor.length() == 1)
-               ? new PrimitiveClassDescImpl(descriptor)
+               ? Wrapper.forPrimitiveType(descriptor.charAt(0)).primitiveClassDescriptor()
```

Comment:
Wait a second, it seems `ClassDesc.ofDescriptor("L")` will return `ConstantDescs.CD_Object` now. Might be good to have a test to verify this doesn't happen.

Reply:
Ignore me, `Wrapper.ofPrimitiveType` doesn't return Object.

Reply:
No, `Wrapper.forPrimitiveType` throws IAE if passed `L`, same exception as today with a slightly different message. 

And yes, if this does not happen there's a test that fails.